apm-lambda-extension/Makefile

```diff
@@ -1,31 +1,96 @@
-build:
+SHELL = /bin/bash -eo pipefail
+
+AWS_FOLDER = .aws
+
 ifndef GOARCH
 	GOARCH=amd64
 endif
+
+export AWS_FOLDER GOARCH
+
+build:
 	GOOS=linux go build -o bin/extensions/apm-lambda-extension main.go
 	chmod +x bin/extensions/apm-lambda-extension
-build-and-publish:
-ifndef AWS_DEFAULT_REGION
-	$(error AWS_DEFAULT_REGION is undefined)
-endif
+build-and-publish: validate-layer-name validate-aws-default-region
```

Comment:
`validate-*` are the goals to avoid duplication of missing arguments/env-variables in some other Make goals